content/blog/2020-03-13-rclean/index.Rmd

```diff
@@ -0,0 +1,465 @@
+main goal of an analysis is the results and taking time to craft
+transparent, dependable software is not the priority. As such, taking
+time to go back through a script and remove code is time
+wasted. Relatedly, for most researchers the best way to keep track of
+the various analyses that they have explored is to keep them in the
+script, as they do not use a rigorous version control system but
+instead rely on file backups and informal versioning. Although we
+can't give researchers more hours in the day, providing an easier and
+more reliable means to remove unused code will lower the barrier to
+creating better, cleaner code. Combined with the increasing use of
+version control systems and digital notebooks, the practice of
+"saving" analytical ideas in a script will become less common and code
+quality will increase.
+
+The process that [Rclean](https://docs.ropensci.org/rclean) uses
+relies on the generation of data provenance.  The term provenance
+means information about the origins of some object. Data provenance is
+a formal representation of the execution of a computational process
+(https://www.w3.org/TR/prov-dm/), to rigorously determine the the
+unique computational pathway from inputs to results [@Carata2014]. To
+avoid confusion, note that "data" in this context is used in a broad
+sense to include all of the information generated during computation,
+not just the data that are collected in a research project that are
+used as input to an analysis. Having the formalized, mathematically
+rigorous representation that data provenance provides guarantees that
+the analyses that [Rclean](https://docs.ropensci.org/rclean) conducts
+are theoretically sound. Most importantly, because the relationships
+defined by the provenance can be represented as a graph, it is
+possible to apply network search algorithms to determine the minimum
+and sufficient code needed to generate the chosen result in the
+`clean` function.
+
+There are multiple approaches to collecting data provenance, but
+[Rclean](https://docs.ropensci.org/rclean) uses "prospective"
+provenance, which analyzes code and uses language specific information
+to predict the relationship among processes and data
+objects. [Rclean](https://docs.ropensci.org/rclean) relies on a
+library called CodeDepends to gather the prospective provenance for
+each script. For more information on the mechanics of the CodeDepends
+package, see [@R-CodeDepends]. To get an idea of what data provenance
+is, we can use `code_graph` function to get a graphical representation
+of the prospective provenance generated for
+[Rclean](https://docs.ropensci.org/rclean) \ref{fig:prov-graph}.
+
+```{R load-script, echo = FALSE, results = "hide"}
+library(CodeDepends)
+library(Rclean)
+```
+
+```{R prov-graph, dpi = 300, fig.cap = "Network diagram of the prospective data provenance generated for an example script. Arrows indicate which lines of code (numbered) produced which objects (named).", echo = FALSE}
```

Comment:
Two comments here:
1) If the numbered lines produce (->) named objects, what does it mean when an object leads (->) to a numbered line? e.g., 1 -> mat -> 2
2) See comments above on using the `hugoopts` chuck option to get the caption and alt text to work properly.

Reply:
I added more text here to explain this: b699bda2.